src/modules/ekf2/test/test_EKF_drag_fusion.cpp

```diff
@@ -0,0 +1,238 @@
+ * 2. Redistributions in binary form must reproduce the above copyright
+ *    notice, this list of conditions and the following disclaimer in
+ *    the documentation and/or other materials provided with the
+ *    distribution.
+ * 3. Neither the name PX4 nor the names of its contributors may be
+ *    used to endorse or promote products derived from this software
+ *    without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+ * FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+ * COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+ * INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+ * BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS
+ * OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED
+ * AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ * LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+ * ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ *
+ ****************************************************************************/
+
+/**
+ * Test the fusion of body frame specific forces for the estimation of wind speed
+ * @author Paul Riseborough <gncsolns@gmail.com>
+ */
+
+#include <gtest/gtest.h>
+#include "EKF/ekf.h"
+#include "sensor_simulator/sensor_simulator.h"
+#include "sensor_simulator/ekf_wrapper.h"
+
+
+class EkfDragFusionTest : public ::testing::Test
+{
+public:
+
+	EkfDragFusionTest(): ::testing::Test(),
+		_ekf{std::make_shared<Ekf>()},
+		_sensor_simulator(_ekf),
+		_ekf_wrapper(_ekf),
+		_quat_sim(Eulerf(0.0f, 0.0f, 0.0f)) {};
+
+	std::shared_ptr<Ekf> _ekf;
+	SensorSimulator _sensor_simulator;
+	EkfWrapper _ekf_wrapper;
+	const Quatf _quat_sim;
+
+	// Setup the Ekf with synthetic measurements
+	void SetUp() override
+	{
+		// run briefly to init, then manually set in air and at rest (default for a real vehicle)
+		_ekf->init(0);
+		_ekf->set_is_fixed_wing(false);
+		_ekf->set_in_air_status(false);
+		_ekf->set_vehicle_at_rest(true);
+	}
+
+	// Use this method to clean up any memory, network etc. after each test
+	void TearDown() override
+	{
+	}
+};
+
+TEST_F(EkfDragFusionTest, testForwardMomentumDrag)
+{
+
+	const float pitch = math::radians(10.0f);
+	const float roll = math::radians(0.0f);
+	const Eulerf euler_angles_sim(roll, pitch, 0.0f);
+	const Quatf quat_sim(euler_angles_sim);
+	_sensor_simulator.simulateOrientation(quat_sim);
+
+	_ekf_wrapper.enableGpsFusion();
+	_sensor_simulator.startGps();
+
+	// Apply parameter changes required to do drag fusion wind estimation
+	parameters *_params = _ekf->getParamHandle();
+	_params->bcoef_x = 0.0f;
+	_params->bcoef_y = 0.0f;
+	_params->mcoef = 0.15f;
+	_params->fusion_mode = 33;
```

Comment:
please use the EkfWrapper class to set and get parameters; especially for the fusion bitmask:
1. to set a bit in the aid mask: https://github.com/PX4/PX4-Autopilot/blob/deb938fceaa79b927ae0f622d8b2b8c4ad10d391/src/modules/ekf2/test/sensor_simulator/ekf_wrapper.cpp#L83-L86
1. to set a value in a parameter: https://github.com/PX4/PX4-Autopilot/blob/deb938fceaa79b927ae0f622d8b2b8c4ad10d391/src/modules/ekf2/test/sensor_simulator/ekf_wrapper.cpp#L98-L101

Reply:
Done